packages/google-pay-integration/src/gateways/google-pay-braintree-gateway.ts

```diff
@@ -101,6 +101,10 @@ export default class GooglePayBraintreeGateway extends GooglePayGateway {
         return nonce;
     }
 
+    async getDeviceSessionId() {
+        return this._getBraintreeDeviceData();
+    }
```

Comment:
This method should be changed to:
```suggestion
    async getExtraPaymentData() {
        return {
            deviceSessionId: await this._getBraintreeDeviceData();
        };
    }
```